libsmtutil/CHCSmtLib2Interface.cpp

```diff
@@ -94,14 +94,10 @@ tuple<CheckResult, Expression, CHCSolverInterface::CexGraph> CHCSmtLib2Interface
 	string accumulated{};
 	swap(m_accumulatedOutput, accumulated);
 	solAssert(m_smtlib2, "");
-	writeHeader();
```

Comment:
Aren't these lines the same as the contents of `dumpQuery`?

Reply:
The same. But we want only a part of this to be stored in `m_accumulatedOutput`.

Reply:
But `m_accumulatedOutput` is thrown away at the end anyway, so I think we could basically use your new function `dumpQuery` here directly, no?

Reply:
Right!

Reply:
@leonardoalt it's ready for another CR.